test/test_csvs.py

```diff
@@ -0,0 +1,253 @@
+# OOIPLACEHOLDER
+#
+# Copyright 2018 Raytheon Co.
+
+import glob
+import io
+import os
+import pandas as pd
+import re
+import unittest
+from nose.plugins.attrib import attr
+
+
+@attr('UNIT')
+class GeneralUnitTest(unittest.TestCase):
+    # declare misc global objects
+    _encoding = 'utf-8'
+
+    _validate_dictionary  = {} # populated in setUp
+    _validation_positions = {'_Array ID': (0, 8),
+                             '_Platform ID': (9, 14),
+                             '_Instrument': (18, 27)
+                            }
+
+    # declare the global validation objects
+    _reference_designators = set()
+    _parameters = set()
+    _units = set()
+
+    @staticmethod
+    def _get_path(file_name):
+        return os.path.join(os.path.dirname(__file__), file_name)
+
+    @staticmethod
+    def _is_empty_or_numeric(value):
+        # if it's empty it's ok
+        if len(value) == 0: return True
+
+        # any integer or float value will succeed
+        try:
+            float(value)
+        except ValueError:
+            return False
+
+        return True
+
+    def setUp(self):
+        # initialize the global objects once
+        if not self._reference_designators:
+            for file_name in glob.glob("*_Deploy.csv"):
+                file = pd.read_csv(self._get_path(file_name), encoding=self._encoding, na_filter=False)
+                header = list(file)
+                # the placement of this column varies from one CSV to another
+                column_index = header.index('Reference Designator')
+                # the "i" variable isn't used but is needed to catch the returned value
+                for i,line in file.iterrows():
+                    self._reference_designators.add(line[column_index])
+
+        if not self._units:
+            unit_regex = re.compile(r'INSERT INTO "unit" VALUES\(\d+,\'(.*)\'\)')
+            with io.open(self._get_path('preload_database.sql'), mode='rt', encoding=self._encoding) as file:
```

Comment:
I'm sure it would be better to use a python database library (e.g. peewee) to parse the sql file, but this works...

Reply:
I'll consider what you're speaking of as a possible future enhancement. For now I'm calling it good.